src/sentry/models/artifactbundle.py

```diff
@@ -255,6 +261,38 @@ def _build_memory_maps(self):
             # Building the map for url lookup.
             self._entries_by_url[info.get("url")] = (file_path, info)
 
+    def extract_debug_ids_from_manifest(
+        self,
+    ) -> Tuple[Optional[str], Set[Tuple[SourceFileType, str]]]:
+        # We use a set, since we might have the same debug_id and file_type.
+        debug_ids_with_types = set()
+
+        # We also want to extract the bundle_id which is also known as the bundle debug_id. This id is used to uniquely
+        # identify a specific ArtifactBundle in case for example of future deletion.
+        #
+        # If no id is found, it means that we must have an associated release to this ArtifactBundle, through the
+        # ReleaseArtifactBundle table.
+        manifest = self.manifest
+        bundle_id = manifest.get("debug_id")
+        if bundle_id is not None:
+            bundle_id = self.normalize_debug_id(bundle_id)
```

Comment:
I believe this is a slightly different concern and would better fit into a dedicated `extract_bundle_id` method.

Reply:
I agree, will try to lift it out!